java/src/org/openqa/selenium/grid/node/relay/RelaySessionFactory.java

```diff
@@ -146,7 +148,13 @@ public Either<WebDriverException, ActiveSession> apply(CreateSessionRequest sess
           new SessionNotCreatedException(
               "New session request capabilities do not " + "match the stereotype."));
     }
-    capabilities = capabilities.merge(stereotype);
+        
+    // remove browserName capability as it breaks appium tests when app is provided, and for any other test, browser should be provided
+    // in session request
+    MutableCapabilities filteredStereotype = new MutableCapabilities(stereotype);
+    filteredStereotype.setCapability(CapabilityType.BROWSER_NAME, (String)null);
+
+    capabilities = capabilities.merge(filteredStereotype);
```

Comment:
Hello @diemol 

That's what I proposed first, but I think it's a more general case. After all, before 4.18.1 release, this capability was not there at all.
I we need a specific browser through relay session (appium or anything else), It will be provided through requested session right ?

Reply:
`browserName` is always taken in to account to match, so we cannot just ignore it. 

In addition, Relay is also used to forward session commands to cloud vendors, and `browserName` is important there.

Reply:
Ok, so I will change my code accordingly